rancher2/resource_rancher2_cluster_v2.go

```diff
@@ -64,6 +70,37 @@ func resourceRancher2ClusterV2() *schema.Resource {
 	}
 }
 
+func resourceRancher2ClusterV2Resource() *schema.Resource {
+	return &schema.Resource{
+		Schema: clusterV2FieldsV0(),
+	}
+}
+
+func resourceRancher2ClusterV2StateV0(rawState map[string]any, meta interface{}) (map[string]any, error) {
+	if rkeConfigs, ok := rawState["rke_config"].([]any); ok && len(rkeConfigs) > 0 {
+		for i := range rkeConfigs {
+			if rkeConfig, ok := rkeConfigs[i].(map[string]any); ok && len(rkeConfig) > 0 {
+				if machineSelectorConfigs, ok := rkeConfig["machine_selector_config"].([]any); ok && len(machineSelectorConfigs) > 0 {
+
+					// upgrade all machine selector configs
+					for m := range machineSelectorConfigs {
+						if machineSelectorConfig, ok := machineSelectorConfigs[m].(map[string]any); ok && len(machineSelectorConfig) > 0 {
+
+							// machine selector config data found. Migrate state from map -> list
+							if config, ok := machineSelectorConfig["config"].([]any); ok {
+								var newValue []any
+								newValue = append(newValue, config)
+								rawState["rke_config"].([]interface{})[i].(map[string]any)["machine_selector_config"].([]any)[m].(map[string]any)["config"] = newValue
+							}
```

Comment:
I may be wrong, but this dosen't cast  from a Map to a list. (at leas not a go map to a go list). 

If "config" is a map it will be ignored. 

If "config" already is a list/slice this function will take place and replace something like:`...config := [1, 2, 3]`  to `config = [[1,2,3]]`

Is this the idea of the function ?

Reply:
Oh actually, I need to update the state migration logic since I adjusted the fix. This should be migration from a map -> string, I'll update that.